ros/src/sensing/filters/packages/points_preprocessor/CMakeLists.txt

```diff
@@ -61,29 +61,41 @@ target_link_libraries(ring_ground_filter
 
 
 # Ray Ground Filter
-add_executable(ray_ground_filter
-    nodes/ray_ground_filter/ray_ground_filter.cpp
-)
+add_library(ray_ground_filter_lib SHARED
+  nodes/ray_ground_filter/ray_ground_filter.cpp)
 
 if (OPENMP_FOUND)
-    set_target_properties(ray_ground_filter PROPERTIES
+    set_target_properties(ray_ground_filter_lib PROPERTIES
         COMPILE_FLAGS ${OpenMP_CXX_FLAGS}
         LINK_FLAGS ${OpenMP_CXX_FLAGS}
     )
 endif()
 
-target_include_directories(ray_ground_filter PRIVATE
+target_include_directories(ray_ground_filter_lib PRIVATE
```

Comment:
This is because the unit test itself is an executable. To my knowledge, there's no way for the unit test executable to reach code from the node executable. As a library, both the unit test executable and the node executable can use the same code.